src/components/Button/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { PropsWithChildren, MouseEvent } from 'react';
+import { styButtonPrimary, styUndefinedBtn } from './styles';
+import { ButtonProps } from './buttonProps';
+import { isValidHttpUrl, Route } from './Route';
+import { cx } from '@emotion/css';
+
+const Button = (props: PropsWithChildren<ButtonProps>): JSX.Element => {
+    const { btnSize, btntype, btnColor, link, element, children, ...rest } =
```

Comment:
```suggestion
    const { btnSize = "large", btntype = "dashed", btnColor = "green", link, element, children, ...rest } =
```

from what i see you've made the variables optional right by adding "?" after the props name in the interface..

if that is your case.. you need to assign the default value here..